lightkurve/correctors/cbvcorrector.py

```diff
@@ -1,258 +1,946 @@
+            **kwargs):
+        """ Class method to instantiate a KeplerCotrendingBasisVectors object
+        from a CBV FITS HDU.
+
+        Kepler/K2 CBVs are all in the same FITS file for each quarter/campaign,
+        so, when instantiating the CBV object we must specify which module and
+        output we desire. Only Single-Scale CBVs are stored for Kepler.
+
+        Parameters
         ----------
-        .. [1] https://en.wikipedia.org/wiki/Bayes_factor
+        HDU : astropy.io.fits.hdu.hdulist.HDUList
+            A pyfits opened FITS file containing the CBVs
+        module : int
+            Kepler CCD module 2 - 84
+        output : int
+            Kepler CCD output 1 - 4
+        **kwargs : Optional arguments
+                    Passed to the TimeSeries superclass
         """
 
-        self.bayes_factor, cost = [], []  # bayes_factor here is actually the
-                                          # negative log of the bayes factor
-        self.correct(cbvs=[1], options={'xtol': 1e-6, 'ftol': 1e-6, 'maxfev': 2000})
-        cost.append(self.opt_result.fun)
-        for n in tqdm(range(2, self._ncbvs+1)):
-            cbv_list = list(range(1, n+1))
-            self.correct(cbv_list, options={'xtol': 1e-6, 'ftol': 1e-6, 'maxfev': 2000})
-            cost.append(self.opt_result.fun)
-            # cost is the negative log of the posterior evaluated at the
-            # Maximum A Posterior Probability (MAP) estimator
-            self.bayes_factor.append((cost[n-2] - cost[n-1]))
-            # so cost[n-2] - cost[n-1] = -log(p1) + log(p2) = log(p2/p1)
-            # where p1 is the posterior probability (evaluated at the MAP)
-            # for the model with n-2 cbvs and p2 is the posterior probability
-            # also evaluated at the MAP for the model with n-1 cbvs
-        k = np.argmin(self.bayes_factor)
-        # transform to get the actual Bayes factor
-        self.bayes_factor = np.exp(-np.array(self.bayes_factor))
-        # the k+2 here comes from the fact that Python indexes begin
-        # from 0 and we count CBVs starting from 1 and also
-        # note that range(1, k) equals the interval [1, k), which excludes k.
-        return list(range(1, k+2))
-
-    def get_cbv_url(self):
-        # gets the html page and finds all references to 'a' tag
-        # keeps the ones for which 'href' ends with 'fits'
-        # this might slow things down in case the user wants to fit 1e3 stars
-        soup = BeautifulSoup(requests.get(self.cbv_base_url).text, 'html.parser')
-        cbv_files = [fn['href'] for fn in soup.find_all('a') if fn['href'].endswith('fits')]
+        assert module > 1 and module < 85, 'Invalid module number'
+        assert output > 0 and output < 5, 'Invalid output number'
+
+        # Get the mission: Kepler or K2
+        # Sadly, the HDU does not explicitly say if this is Kepler or K2 CBVs.
+        if HDU['PRIMARY'].header.__contains__('QUARTER'):
```

Comment:
The `__contains__` method implements Python's `in` operator, so this is exactly the same but more Pythonic:
```suggestion
        if 'QUARTER' in HDU['PRIMARY'].header:
```

Reply:
Changed.